src/contracts/core/StakeRootCompendium.sol

```diff
@@ -234,226 +202,155 @@ contract StakeRootCompendium is StakeRootCompendiumStorage {
-            "StakeRootCompendium._setProofInterval: make sure there are no pending proofs"
+            avsDirectory.isOperatorSet(operatorSet.avs, operatorSet.operatorSetId),
+            "StakeRootCompendium.depositForOperatorSet: operator set does not exist"
         );
-        _updateCumulativeCharges();
-        proofInterval = proofInterval;
+        depositInfo[operatorSet.avs][operatorSet.operatorSetId].balance += uint96(msg.value);
     }
 
-    function _updateCumulativeCharges() internal {
-        uint32 latestCalculationTimestamp = _getLatestCalculationTimestamp();
-        (uint224 currentConstantCumulativeCharge, uint224 currentLinearCumulativeCharge) = _getCurrentCumulativeCharges(latestCalculationTimestamp);
-
-        // update the cumulative charge snapshots
-        cumulativeContantChargeSnapshot.push(
-            latestCalculationTimestamp,
-            currentConstantCumulativeCharge
+    function _addToStakeTree(IAVSDirectory.OperatorSet memory operatorSet) internal {
+        require(
+            operatorSetToOneBasedIndex[operatorSet.avs][operatorSet.operatorSetId] == 0, 
+            "StakeRootCompendium._addToStakeTree: operator set already in stake tree"
         );
-        cumulativeLinearChargeSnapshot.push(
-            latestCalculationTimestamp,
-            currentLinearCumulativeCharge
+        require(
+            operatorSets.length < MAX_NUM_OPERATOR_SETS,
+            "StakeRootCompendium.setStrategiesAndMultipliers: too many operator sets"
         );
-    }
-
-    function _getCurrentCumulativeCharges(uint32 latestCalculationTimestamp) internal view returns(uint224, uint224) {
-        return (
-            _getCurrentCumulativeCharge(cumulativeContantChargeSnapshot, constantChargePerProof, latestCalculationTimestamp),
-            _getCurrentCumulativeCharge(cumulativeLinearChargeSnapshot, linearChargePerProof, latestCalculationTimestamp)
+        // min deposit balance   
+        require(
+            depositInfo[operatorSet.avs][operatorSet.operatorSetId].balance >= MIN_DEPOSIT_BALANCE,
+            "StakeRootCompendium._addToStakeTree: operator set does not meet minimum deposit balance"
         );
+        operatorSets.push(operatorSet);
+        // note: operatorSetIndexPlusOne is 1 indexed
+        operatorSetToOneBasedIndex[operatorSet.avs][operatorSet.operatorSetId] = uint208(operatorSets.length);
     }
 
-    function _getPreviousCumulativeCharges(uint32 timestamp) internal view returns(uint224, uint224) {
-        return (
-            cumulativeContantChargeSnapshot.upperLookup(timestamp),
-            cumulativeLinearChargeSnapshot.upperLookup(timestamp)
+    function _removeOperatorSetFromStakeTree(IAVSDirectory.OperatorSet memory operatorSet) internal {
+        uint256 operatorSetOneBasedIndex = operatorSetToOneBasedIndex[operatorSet.avs][operatorSet.operatorSetId];
```

Comment:
we need to update the totalCharge in this function in case this is a force removal that isn't a removal of strategies